app/Filament/Pages/EditProfile.php

```diff
@@ -146,19 +146,45 @@ public function form(Form $form): Form
         ])->map(
             fn ($day) => Grid::make(3)
                 ->schema([
-                    Toggle::make("office_hours_days.{$day}.enabled")
+                    Toggle::make("office_hours.{$day}.enabled")
                         ->label(str($day)->ucfirst())
                         ->inline(false)
                         ->live(),
-                    TimePicker::make("office_hours_days.{$day}.start")
-                        ->visible(fn (Get $get) => $get("office_hours_days.{$day}.enabled")),
-                    TimePicker::make("office_hours_days.{$day}.end")
-                        ->visible(fn (Get $get) => $get("office_hours_days.{$day}.enabled")),
+                    TimePicker::make("office_hours.{$day}.starts_at")
+                        ->required()
+                        ->visible(fn (Get $get) => $get("office_hours.{$day}.enabled")),
+                    TimePicker::make("office_hours.{$day}.ends_at")
+                        ->required()
+                        ->visible(fn (Get $get) => $get("office_hours.{$day}.enabled")),
                 ])
         )->toArray();
 
         return $form
             ->schema([
+                Section::make('Public Profile')
+                    ->aside()
+                    ->schema([
+                        Toggle::make('has_enabled_public_profile')
+                            ->label('Enable public profile')
+                            ->live(),
+                        TextInput::make('public_profile_slug')
+                            ->label('Url')
+                            ->visible(fn (Get $get) => $get('has_enabled_public_profile'))
+                            //TODO: default doesn't work for some reason
+                            ->afterStateHydrated(fn (TextInput $component, $state) => $component->state($state ?? str($user->name)->lower()->slug('')))
+                            ->unique(ignoreRecord: true)
+                            ->maxLength(255)
+                            ->required()
+                            ->prefix(str(route('view-public-user-profile', ['user' => $user->id]))->beforeLast('/')->append('/'))
+                            ->suffixAction(
+                                FormAction::make('viewPublicProfile')
+                                    ->url(fn (Get $get) => route('view-public-user-profile', ['user' => $get('public_profile_slug')]))
+                                    ->icon('heroicon-m-arrow-top-right-on-square')
+                                    ->openUrlInNewTab()
+                                    ->visible(fn (Get $get) => $get('public_profile_slug')),
+                            )
```

Comment:
Is it possible for this to be hidden until the profile is saved and the slug is set? If I just created or changed my profile and then I click this it brings me to the 404 page.